model/mongo/src/main/java/org/keycloak/models/mongo/keycloak/adapters/UserAdapter.java

```diff
@@ -177,31 +183,73 @@ public void setTotp(boolean totp) {
 
     @Override
     public void updateCredential(UserCredentialModel cred) {
+
+        if (cred.getType().equals(UserCredentialModel.PASSWORD)) {
+            updatePasswordCredential(cred);

```

Comment:
Rather than duplicating the updatePasswordCredential method in all stores we should rename CredentialValidation to CredentialUtils and add it as a static method there